spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/task/impl/DefaultDataFlowTaskExecutionQueryDao.java

```diff
@@ -509,4 +521,25 @@ private List<String> getTaskArguments(long taskExecutionId) {
 				handler);
 		return params;
 	}
+
+	@Override
+	public void populateCtrStatus(Collection<ThinTaskExecution> taskExecutions) {
+		final AtomicInteger updated = new AtomicInteger(0);
+		Map<Long, ThinTaskExecution> taskExecutionMap = taskExecutions.stream()
+				.collect(Collectors.toMap(ThinTaskExecution::getExecutionId, Function.identity()));
+		String ids = taskExecutionMap.keySet()
+				.stream()
+				.map(Object::toString)
+				.collect(Collectors.joining(","));
+		String sql = FIND_CTR_STATUS.replace(":taskExecutionIds", ids);
+		jdbcTemplate.query(sql, rs -> {
+			Long id = rs.getLong("TASK_EXECUTION_ID");
+			String ctrStatus = rs.getString("CTR_STATUS");
+			logger.debug("populateCtrStatus:{}={}", id, ctrStatus);
+			ThinTaskExecution execution = taskExecutionMap.get(id);
+			Assert.notNull(execution, "Expected TaskExecution for " + id + " from " + ids);
+			updated.incrementAndGet();
```

Comment:
Not sure why we need the AtomicInteger here.   I don't see it used.

Reply:
it was logged previously. removed now.